src/content-tags-drawer/ContentTagsDropDownSelector.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect } from 'react';
+import {
+  SelectableBox,
+  Icon,
+  Spinner,
+} from '@edx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { ArrowDropDown, ArrowDropUp } from '@edx/paragon/icons';
+import PropTypes from 'prop-types';
+import messages from './messages';
+import './ContentTagsDropDownSelector.scss';
+
+import { useTaxonomyTagsDataResponse, useIsTaxonomyTagsDataLoaded } from './api/hooks/selectors';
+
+const ContentTagsDropDownSelector = ({
+  taxonomyId, level, subTagsUrl,
+}) => {
+  const intl = useIntl();
+  // This array represents the states of the dropdowns on this level
+  // Each index contains whether the dropdown is open/closed
+  // O === Closed
+  // 1 === Open
+  const [dropdownStates, setDropdownStates] = useState([]);
```

Comment:
I think some of this code could be simplified if this an object mapping the index to true/false/undefined. 

isOpen can stay the same. 
Updating an index can be `setDropDownStates({...dropdownState, i:!dropdownState[i]})`
You don't need to initialize since initial keys are undefined which are falsy.